# Upstream status: Already there, needed to fix build of kernel/ptrace.c
# Author: Kyle McMartin <kyle@debian.org>, poached from my commit to cvs

--- a/include/asm-parisc/spinlock.h	20 Oct 2005 16:07:04 -0000	1.17
+++ b/include/asm-parisc/spinlock.h	25 Mar 2006 04:22:02 -0000	1.18
@@ -134,14 +134,22 @@ static  __inline__ int __raw_write_trylo
 	return 1;
 }
 
-static __inline__ int __raw_is_read_locked(raw_rwlock_t *rw)
+/*
+ * read_can_lock - would read_trylock() succeed?
+ * @lock: the rwlock in question.
+ */
+static __inline__ int __raw_read_can_lock(raw_rwlock_t *rw)
 {
-	return rw->counter > 0;
+	return rw->counter >= 0;
 }
 
-static __inline__ int __raw_is_write_locked(raw_rwlock_t *rw)
+/*
+ * write_can_lock - would write_trylock() succeed?
+ * @lock: the rwlock in question.
+ */
+static __inline__ int __raw_write_can_lock(raw_rwlock_t *rw)
 {
-	return rw->counter < 0;
+	return !rw->counter;
 }
 
 #endif /* __ASM_SPINLOCK_H */
